dht/token-sharding.hh

```diff
@@ -67,7 +73,12 @@ public:
      *   }
      *
      */
-    virtual unsigned shard_for_reads(const token& t) const = 0;
+    unsigned shard_for_reads(const token& t) const {
+        // FIXME: Consider throwing when there is no owning shard on the current host rather than returning 0.
+        // It's a coordination mistake to route requests to non-owners. Topology coordinator should synchronize
+        // with request coordinators before moving the shard away.
```

Comment:
Isn't it a driver mistake? Or just the driver nor having an up-to-date replica map?

We can consider returning an error (not throwing) if we can negotiate a leader-following protocol.

Reply:
> Isn't it a driver mistake? Or just the driver nor having an up-to-date replica map?

Any node in a cluster can be an LWT coordinator. It's better if the driver talks to a replica, but it's more of an optimization not the guarantee.

Reply:
Or just no prepared statement from (cqlsh for instance).